src/test/java/uk/gov/hmcts/cmc/claimstore/events/offer/AgreementCounterSignedCitizenActionsHandlerTest.java

```diff
@@ -0,0 +1,114 @@
+package uk.gov.hmcts.cmc.claimstore.events.offer;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.hmcts.cmc.claimstore.config.properties.notifications.EmailTemplates;
+import uk.gov.hmcts.cmc.claimstore.config.properties.notifications.NotificationTemplates;
+import uk.gov.hmcts.cmc.claimstore.config.properties.notifications.NotificationsProperties;
+import uk.gov.hmcts.cmc.claimstore.services.notifications.OfferMadeNotificationService;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.offers.MadeBy;
+import uk.gov.hmcts.cmc.domain.models.offers.Settlement;
+import uk.gov.hmcts.cmc.domain.models.sampledata.SampleClaim;
+
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AgreementCounterSignedCitizenActionsHandlerTest {
```

Comment:
That test is very similar to `SettlementReachedCitizenNotificationServiceTest` which actually tests a bit more. I would like to remove `AgreementCounterSignedCitizenActionsHandlerTest` and have integration tests that tests an application from HTTP request side and makes sure `AgreementCounterSignedCitizenActionsHandler` is called with right event when offer got counter signed.

Reply:
done